modules/postgresql/variables.tf

```diff
@@ -439,3 +439,28 @@ variable "data_cache_enabled" {
   type        = bool
   default     = false
 }
+
+variable "psc_consumer" {
+  description = "The psc consumer to be created on the same project as the SQL instance(s). Remember to add the project under psc_allowed_consumer_projects in the ip_configuration block."
+  type = object({
+    subnet_id               = optional(string, "")
+    network_id              = optional(string, "")
+    enabled                 = optional(bool, false)
+    allow_psc_global_access = optional(bool, false)
+  })
+
+  default = {}
+
+  validation {
+    condition     = (!var.psc_consumer.enabled || (var.psc_consumer.network_id != "" && var.psc_consumer.subnet_id != ""))
```

Comment:
Shouldn't this be based on the error message:
```
var.psc_consumer.enabled && (var.psc_consumer.network_id != "" || var.psc_consumer.subnet_id != "")
```

Reply:
Your expression would return true ( valid value ) in case enabled is true and one of network_id or subnet_id is null ( 1 && ( 1 || 0 )). Our expression would return true with the default object and would return true if enabled is true and both network_id or subnet_id are populated. If we "turn off" the module by setting enabled == false the expression would also be true ( expected behaviour ) irrespectively of the input data.